### ISSUE TITLE
[Bug]: Storybook won't start and prints confusing messages

### ISSUE THREAD

**Issue by kaiyoma** [storybookjs/storybook#22781]
### Describe the bug

I'm trying to migrate from Storybook 6.5 to 7.0 and I can't get Storybook to start correctly. I've already followed the entire migration guide. The dev server seems to start, but ends with an error:

```
info => Starting manager..
╭──────────────────────────────────────────────────╮
│                                                  │
│   Storybook 7.0.17 for react-vite started        │
│   832 ms for manager and 1.1 min for preview     │
│                                                  │
│    Local:            http://localhost:9001/      │
│    On your network:  http://10.95.67.95:9001/    │
│                                                  │
╰──────────────────────────────────────────────────╯
WARN 🚨 SyntaxError: Unexpected token, expected "}" (19:10)
```

There's no further information and no filename is printed, so I have no idea where this is happening. And the dev server doesn't seem to support any kind of debug logging, so I'm stuck here.

If I build Storybook and enable the highest level of logging, I get this:

```
info => Loading presets
info => Building manager..
info => Manager built (982 ms)
info => Copying static files: C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@storybook+manager@7.0.17\node_modules\@storybook\manager\static at C:\Users\kgetz\Work\event-viewer\dist\storybook\sb-common-assets
WARN 🚨 SyntaxError: Unexpected token, expected "}" (19:10)
ERR! SyntaxError: Unexpected token, expected "}" (19:10)
ERR!     at instantiate (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:653:32)
ERR!     at constructor (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:947:12)
ERR!     at TypeScriptParserMixin.raise (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:3271:19)
ERR!     at TypeScriptParserMixin.unexpected (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:3301:16)
ERR!     at TypeScriptParserMixin.expect (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:3643:28)
ERR!     at TypeScriptParserMixin.jsxParseExpressionContainer (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:6873:10)
ERR!     at TypeScriptParserMixin.jsxParseElementAt (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:6947:36)
ERR!     at TypeScriptParserMixin.jsxParseElement (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:6991:17)
ERR!     at TypeScriptParserMixin.parseExprAtom (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:7003:19)
ERR!     at TypeScriptParserMixin.parseExprSubscripts (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:10880:23)
ERR!  SyntaxError: Unexpected token, expected "}" (19:10)
ERR!     at instantiate (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:653:32)
ERR!     at constructor (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:947:12)
ERR!     at TypeScriptParserMixin.raise (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:3271:19)
ERR!     at TypeScriptParserMixin.unexpected (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:3301:16)
ERR!     at TypeScriptParserMixin.expect (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:3643:28)
ERR!     at TypeScriptParserMixin.jsxParseExpressionContainer (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:6873:10)
ERR!     at TypeScriptParserMixin.jsxParseElementAt (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:6947:36)
ERR!     at TypeScriptParserMixin.jsxParseElement (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:6991:17)
ERR!     at TypeScriptParserMixin.parseExprAtom (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:7003:19)
ERR!     at TypeScriptParserMixin.parseExprSubscripts (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\@babel+parser@7.21.4\node_modules\@babel\parser\lib\index.js:10880:23) {
ERR!   code: 'BABEL_PARSER_SYNTAX_ERROR',
ERR!   reasonCode: 'UnexpectedToken',
ERR!   loc: Position { line: 19, column: 10, index: 654 },
ERR!   pos: [Getter/Setter]
ERR! }
WARN 🚨 SyntaxError: Unexpected token, expected "}" (19:10)
```

Again, still no filename or text snippet. The build seems to march on, but then I start hitting more problems:

```
./sb-common-assets/fonts.css doesn't exist at build time, it will remain unchanged to be resolved at runtime
transforming (1) iframe.html
./sb-common-assets/fonts.css doesn't exist at build time, it will remain unchanged to be resolved at runtime
✓ 15 modules transformed.
✓ built in 1m 43s
Unexpected token (Note that you need plugins to import files that are not JavaScript)
file: C:/Users/kgetz/Work/event-viewer/node_modules/.pnpm/@storybook+react@7.0.17_ztlwkigqdqeug3eyehxwmcwkt4/node_modules/@storybook/react/dist/chunk-JWY6Y6NU.mjs:1:9
1: import { global } from '@storybook/global';
            ^
2: import React, { StrictMode, Fragment, Component } from 'react';
3: import { unmountElement, renderElement } from '@storybook/react-dom-shim';
✓ 12 modules transformed.
✓ built in 1m 27s
Unexpected token (Note that you need plugins to import files that are not JavaScript)
file: C:/Users/kgetz/Work/event-viewer/node_modules/.pnpm/@storybook+react@7.0.17_ztlwkigqdqeug3eyehxwmcwkt4/node_modules/@storybook/react/dist/chunk-JWY6Y6NU.mjs:1:9
1: import { global } from '@storybook/global';
            ^
2: import React, { StrictMode, Fragment, Component } from 'react';
3: import { unmountElement, renderElement } from '@storybook/react-dom-shim';
transforming (52) common\temp\node_modules\.pnpm\object-assign@4.1.1\node_modules\object-assign
```

### System

```shell
Environment Info:

  System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
  Binaries:
    Node: 16.19.1 - C:\Program Files\Node.js\node.EXE
    npm: 9.6.2 - C:\Program Files\Node.js\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.1702.0), Chromium (113.0.1774.50)
  npmPackages:
    @storybook/addon-actions: 7.0.17 => 7.0.17
    @storybook/addon-controls: 7.0.17 => 7.0.17
    @storybook/addon-docs: 7.0.17 => 7.0.17
    @storybook/addons: 7.0.17 => 7.0.17
    @storybook/api: 7.0.17 => 7.0.17
    @storybook/cli: 7.0.17 => 7.0.17
    @storybook/components: 7.0.17 => 7.0.17
    @storybook/core-events: 7.0.17 => 7.0.17
    @storybook/react: 7.0.17 => 7.0.17
    @storybook/react-vite: 7.0.17 => 7.0.17
    @storybook/theming: 7.0.17 => 7.0.17
```

### Additional context

`.storybook/main.ts`:

```
import type { StorybookConfig } from '@storybook/react-vite';

export default <StorybookConfig>{
  // The addons we want (order matters here)
  addons: ['@storybook/addon-docs', '@storybook/addon-controls', '@storybook/addon-actions'],

  framework: {
    name: '@storybook/react-vite',
  },

  stories: ['../src/containers/**/*.stories.tsx'],
};
```

`.storybook/preview.ts`:

```
import { ArgsTable, Description, Title } from '@storybook/addon-docs';
import type { Preview } from '@storybook/react';
import React from 'react';

import 'antd/dist/antd.less';
import 'draft-js/dist/Draft.css';
import 'react-datetime/css/react-datetime.css';
import 'react-diff-view/style/index.css';
import 'react-grid-layout/css/styles.css';
import 'react-perfect-scrollbar/dist/css/styles.css';
import 'react-resizable/css/styles.css';
import 'react-virtualized/styles.css';
import 'react-vis/dist/style.css';

import '@arista/cv-components/index.css';
import 'geiger-components-new/index.css';

export default <Preview>{
  controls: {
    sort: 'requiredFirst',
  },
  docs: {
    page: () =>
      React.createElement(
        'div',
        null,
        React.createElement(Title),
        React.createElement(Description),
        React.createElement(ArgsTable),
      ),
  },
  options: {
    storySort: (a, b) => {
      return a.title === b.title ? 0 : a.id.localeCompare(b.id, undefined, { numeric: true });
    },
  },
};
```

**Comment by kaiyoma** [comment]
Ahhh, I figured out the syntax error. It's this notation that Babel apparently doesn't understand:

```
export default <Preview>{
  ...
}
```

Changing to this fixing it:

```
const preview: Preview = {
  ...
};

export default preview;
```

Fixing that makes the syntax error go away, but Storybook still isn't starting. The dev server shows a loading indicator that never goes away, and nothing of note in the browser console. A Storybook build encounters the same error about "unexpected token":

```
Unexpected token (Note that you need plugins to import files that are not JavaScript)
file: C:/Users/kgetz/Work/event-viewer/node_modules/.pnpm/@storybook+react@7.0.17_ztlwkigqdqeug3eyehxwmcwkt4/node_modules/@storybook/react/dist/chunk-JWY6Y6NU.mjs:1:9
1: import { global } from '@storybook/global';
            ^
2: import React, { StrictMode, Fragment, Component } from 'react';
3: import { unmountElement, renderElement } from '@storybook/react-dom-shim';
ERR! RollupError: Unexpected token (Note that you need plugins to import files that are not JavaScript)
```

All the while, I can see in Task Manager that Node is hammering my CPU, while everything seems frozen. Is there something really silly and obvious that I'm missing here? I had to do the migration manually because the command-line upgrade tool had the same "hanging" problem.

**Comment by shilman** [comment]
@yannbf here's another one for the "better error messages" pile.

@kaiyoma that sounds super frustrating. are you doing anything custom in your vite config? do you have a babel config? (shouldn't be necessary with vite, but you never know)

**Comment by kaiyoma** [comment]
@shilman I think I managed to get past this hurdle. Thanks for bringing up the Babel config. I still had an old Babel config file laying around (from our previous webpack installation) and still had a lot of Babel/webpack dependencies, but once I nuked all of them, I was able to make some progress with the Storybook build. I'm still seeing error messages and an OOM crash, but I'll dive into those individually.
